llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2590,7 +2591,8 @@ class VPReductionRecipe : public VPSingleDefRecipe {
   VPReductionRecipe(const unsigned char SC, const RecurrenceDescriptor &R,
                     Instruction *I, ArrayRef<VPValue *> Operands,
                     VPValue *CondOp, bool IsOrdered)
-      : VPSingleDefRecipe(SC, Operands, I), RdxDesc(R), IsOrdered(IsOrdered) {
+      : VPSingleDefRecipe(SC, Operands, I, I->getDebugLoc()), RdxDesc(R),
```

Comment:
Better to pass debug loc independent of underlying instruction & update  `clone`, same for  above. Then https://github.com/llvm/llvm-project/pull/120053 won't be needed

Reply:
Updated, thanks.